pkg/registry/registry_quay.go

```diff
@@ -50,6 +66,11 @@ var nextLinkRE = regexp.MustCompile(`^ *<?([^;>]+)>? *(?:;[^;]*)*; *rel="?next"?
 // GetQuayTagsWithDetails fetches tags for the given image including extra information defined in a QuayTag
 // This uses the Quay API, so assumes a Quay host. Other hosts are likely to fail.
 func (r *Registry) GetQuayTagsWithDetails(image string) (tags []QuayTag, err error) {
+	if r.host == "registry-intl.cn-shanghai.aliyuncs.com" {
```

Comment:
Something in the logic seems wonky if this is needed - why not create the correct registry struct right away?

Reply:
Yep, this is quick and dirty just to get unblocked. Moving to separate backend-aware registry behavior is the likely goal but it's a bigger change and will take me a bit longer